pkg/kubectl-argo-rollouts/cmd/set/set_image.go

```diff
@@ -0,0 +1,109 @@
+package set
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/argoproj/argo-rollouts/pkg/apis/rollouts/v1alpha1"
+	rolloutclient "github.com/argoproj/argo-rollouts/pkg/client/clientset/versioned/typed/rollouts/v1alpha1"
+	"github.com/argoproj/argo-rollouts/pkg/kubectl-argo-rollouts/options"
+)
+
+const (
+	setImageExample = `
+  # Set rollout image
+  %[1]s set image my-rollout www=image:v2
+`
+)
+
+const (
+	maxAttempts = 5
+)
+
+// NewCmdSetImage returns a new instance of an `rollouts set image` command
+func NewCmdSetImage(o *options.ArgoRolloutsOptions) *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:          "image ROLLOUT CONTAINER=IMAGE",
+		Short:        "Update the image of a rollout",
+		Example:      o.Example(setImageExample),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) != 2 {
+				return o.UsageErr(c)
+			}
+			rollout := args[0]
+			imageSplit := strings.Split(args[1], "=")
+			if len(imageSplit) != 2 {
+				return o.UsageErr(c)
+			}
+			container := imageSplit[0]
+			image := imageSplit[1]
+
+			rolloutIf := o.RolloutsClientset().ArgoprojV1alpha1().Rollouts(o.Namespace())
+			for attempt := 0; attempt < maxAttempts; attempt++ {
+				err := setImage(rolloutIf, rollout, container, image)
+				if err != nil {
+					if k8serr.IsConflict(err) && attempt < maxAttempts {
+						continue
+					}
+					return err
+				}
+				break
+			}
+			fmt.Fprintf(o.Out, "rollout \"%s\" image updated\n", rollout)
+			return nil
+		},
+	}
+	o.AddKubectlFlags(cmd)
+	return cmd
+}
+
+func setImage(rolloutIf rolloutclient.RolloutInterface, rollout string, container string, image string) error {
+	ro, err := rolloutIf.Get(rollout, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	newRo, err := newRolloutSetImage(ro, container, image)
+	if err != nil {
+		return err
+	}
+	_, err = rolloutIf.Update(newRo)
```

Comment:
I used update and not patch, because strategic merge cause the entire container list to become replaced with a single element. This is also why I retry Conflict errors.